v2/src/modules/search/ModuleSearch.vue

```diff
@@ -321,6 +321,28 @@ const tokensResult = computed(() => {
                     flag: marketData !== false
                 }
             })
+        // Check if hash is of type TOKEN_HASH and get token from tokenMarketInfo
```

Comment:
you can actually move this in front of all the logic, since it doesn't need to do any of the filtering in the market or tokens search arrays. It will speed up function execution.